src/cart/crt_group.c

```diff
@@ -3976,31 +4139,62 @@ int crt_group_rank_remove(crt_group_t *group, d_rank_t rank)
 	}
 	D_RWLOCK_UNLOCK(&grp_priv->gp_rwlock);
 
+	/* Go through associated secondary groups and remove rank from them */
+	if (grp_priv->gp_primary) {
+
+		for (i = 0; i < CRT_MAX_SEC_GRPS; i++) {
+			sec_priv = grp_priv->gp_priv_sec[i];
+			if (sec_priv == NULL)
+				continue;
+
+			rlink = d_hash_rec_find(&sec_priv->gp_p2s_table,
+						(void *)&rank, sizeof(rank));
+			if (!rlink)
+				continue;
+
+			rm = crt_rm_link2ptr(rlink);
+			rc = crt_group_rank_remove(&sec_priv->gp_pub,
+						rm->rm_value);
+
+			if (rc != 0) {
+				D_ERROR("crt_group_rank_remove(%s,%d) failed; "
+					"rc=%d\n",
+					sec_priv->gp_pub.cg_grpid, rm->rm_value,
+					rc);
+			}
+
+			d_hash_rec_decref(&sec_priv->gp_p2s_table, rlink);
+		}
+	}
+
 	if (rc == 0)
 		crt_swim_rank_del(grp_priv, rank);
 out:
 	return rc;
 }
 
 
-int crt_group_info_get(crt_group_t *group, d_iov_t *grp_info)
+int
+crt_group_info_get(crt_group_t *group, d_iov_t *grp_info)
+{
+	D_ERROR("API is currently not supported\n");
+	return -DER_INVAL;
```

Comment:
-DER_NOSYS presumably?  You could add error (NULL) checking to this function however.

Reply:
Will change to -DER_NOSYS